src/algebra/order/monoid_lemmas_zero_lt.lean

```diff
@@ -664,8 +671,35 @@ lemma left.mul_lt_one_of_lt_of_lt [pos_mul_strict_mono α]
   (ha : a < 1) (hb : b < 1) (a0 : 0 < a) : a * b < 1 :=
 mul_lt_of_lt_of_lt_one ha hb a0
 
-/-! Lemmas of the form `b ≤ c → 1 ≤ a → 0 < c → b ≤ c * a`,
-which assume left covariance. -/
+-- proven with `a0 : 0 ≤ a` and `c0 : 0 ≤ c` as `mul_le_of_le_of_le_one'`
+lemma preorder.mul_le_of_le_of_le_one' [pos_mul_mono α] [mul_pos_mono α]
+  (bc : b ≤ c) (ha : a ≤ 1) (a0 : 0 < a) (c0 : 0 < c) : b * a ≤ c :=
+calc  b * a ≤ c * a : mul_le_mul_right' bc a0
+        ... ≤ c * 1 : mul_le_mul_left' ha c0
+        ... = c     : mul_one c
+
+-- proven with `c0 : 0 ≤ c` as `mul_lt_of_lt_of_le_one'`
```

Comment:
Then why do we need this weaker version? Same with all the other lemmas which are strictly weaker versions of lemmas we already have.